options/posix/generic/netdb.cpp

```diff
@@ -290,8 +290,9 @@ struct hostent *gethostbyname(const char *name) {
 	struct mlibc::lookup_result buf;
 	frg::string<MemoryAllocator> canon{getAllocator()};
 	int ret = 0;
-	if ((ret = mlibc::lookup_name_hosts(buf, name, canon, AF_UNSPEC)) <= 0)
-		ret = mlibc::lookup_name_dns(buf, name, canon, AF_UNSPEC);
+	if ((ret = mlibc::lookup_name_ip(buf, name, AF_UNSPEC)) <= 0)
```

Comment:
The use of `AF_UNSPEC` here is incorrect, as we should only ever deal with IPv4 addresses here, so `AF_INET`.

This behavior seems to be a glibc (?) extension, as POSIX 2001 (last issue before this function got removed) states:
> The name argument of gethostbyname() shall be a node name; the behavior of gethostbyname() when passed a numeric address string is unspecified. For IPv4, a numeric address string shall be in the dotted-decimal notation described in inet_addr().

I am personally a bit torn on whether we should default to the glibc extension behavior or whether we should guard that behind the glibc option.